### PR TITLE
docs: updated storybook and doc for compact property

### PR DESCRIPTION
**Pull request by kat10140**
**Related Issue:** #874 

## Summary
fixing the docs, storybooks and demos around the compact property since we are deprecating it.

### CODE REVIEW

File: src/components/calcite-pick-list-item/calcite-pick-list-item.tsx [hunk 1]
```diff
@@ -29,7 +29,7 @@ export class CalcitePickListItem {
   // --------------------------------------------------------------------------
 
   /**
-   * Compact removes the selection icon (radio or checkbox) and adds a compact attribute.
+   * ** DEPRECATED ** Compact removes the selection icon (radio or checkbox) and adds a compact attribute.
    * This allows for a more compact version of the `calcite-pick-list-item`.
```
Comment (driskull) [inline src/components/calcite-pick-list-item/calcite-pick-list-item.tsx]: Needs the `@deprecated` here instead of `**`

Comment (driskull) [reply src/components/calcite-pick-list-item/calcite-pick-list-item.tsx]: See the other deprecated examples.

Comment (driskull) [reply src/components/calcite-pick-list-item/calcite-pick-list-item.tsx]: Applies to other uses of deprecated below.

Comment (driskull) [reply src/components/calcite-pick-list-item/calcite-pick-list-item.tsx]: https://github.com/Esri/calcite-app-components/blob/14361950c8b116a04c7609058fac1045842c97ea/src/components/calcite-block-section/calcite-block-section.tsx#L50

Comment (kat10140) [reply src/components/calcite-pick-list-item/calcite-pick-list-item.tsx]: oh glob! ok will fix

**Pull request merged**